### DISCUSSION TITLE
About leadfield

### DISCUSSION THREAD

**Discussion by lingoodluck** [simnibs/simnibs#463] [8. Simulation workflows]
I downloaded an open-source code that can optimize target areas. The input part of the code uses the position information from the simnibs-generated HDF5 file ['mesh_leadfield']['nodes']['node_coord'] and the data from ['mesh_leadfield']['leadfields']['tdcs_leadfield']. I tried to generate a head model using T1 and T2 MRI from the HCP dataset and then create an HDF5 file. However, when I run the code, it shows that there are no nodes in the region of interest. The area I want to optimize is a spherical region with a center at (47, -13, 52) and a radius of 10mm, which indicates that there are no nodes close to this coordinate in ['mesh_leadfield']['nodes']['node_coord']. However, when I use the ernie head model to generate the HDF5 file, the code runs successfully, and there are nodes in that area in the HDF5 file. I want to know if this issue is due to a problem in generating the head model or in creating the HDF5 file. Do you have any suggestions?

The command I used to generate the head model is:
```
charm 100307 100307_strc_T1w_MPR2_reorient_sformMod.nii 100307_strc_T2w_SPC2_reorient_sformMod.nii --forceqform
```
(I added --forceqform because the qform and sform matrices do not match.)

The code to generate the leadfield is:

```
from simnibs import sim_struct, run_simnibs
tdcs_lf = sim_struct.TDCSLEADFIELD()

tdcs_lf.fnamehead = './100307.msh'
tdcs_lf.pathfem = './leadfield/'

run_simnibs(tdcs_lf)
```

The code to generate the input data is:

```
import h5py 
import numpy as np

f = h5py.File("./leadfield/100307_leadfield_EEG10-10_UI_Jurak_2007.hdf5",'r') 
pos = f['mesh_leadfield']['nodes']['node_coord']
np.save('pos_100307.npy',pos)

lfm = f['mesh_leadfield']['leadfields']['tdcs_leadfield']
np.save('lfm_100307.npy',lfm)
```

**Comment by TorgeW** [comment]
The position that works in Ernie might not work on your other subject. Please visualize the position in your other subject and check if it is where youw ant it to be. Please share this visualization if you have further questions.

**Reply by lingoodluck** [reply]
However, I am using MNI coordinates, and for each subject, I performed the coordinate transformation using the mni2subject_coords command.

**Reply by TorgeW** [reply]
It would still be usefull to show the MNI coordinate, and then also visualize the positions on your subjects so that we can see if the transformation worked correctly. 
Please for the letter use the T1.nii.gz 2from inside the m2m folder of that subject.

**Reply by lingoodluck** [reply]
Thank you for your patient response. I noticed that when generating the leadfield, it defaults to interpolating to the gray matter surface, which may have some impact on the deeper regions. I have chosen to set map_to_surf to False and will try again.
